### ISSUE TITLE
Some Orbitrap files fail to compute matches

### ISSUE THREAD

**Issue by oskarsari** [ultra-trace-systems/mascope#410]
Fail at least with the Mascope version currently deployed at the Orbitrap server. Should verify if the bugs persist at the current head of develop.

`KORBI2_20240624_1230_blank_10_-`
`KORBI2_20240624_1528_Blank_13_-`

**Comment by ghost** [comment]
The issue persists in the latest Mascope version. It's happening in Orbitrap spectra only and is related to the segmentation.

We may want to fit some specific MZ values (278, 287). And it may happen that non-zero segments containing those values do not exist. In those cases, we have `IndexError` in the `segment_spec()` function because `non_zero_indices` is empty but we try to split it, concatenate, etc. We need to add a check if `non_zero_indices` is empty: 

 ```
def segment_spec(sum_spec):
    """Perform segmentation of Orbitrap spectrum

    :param sum_spec: sum of spectra along time dimension
    :type sum_spec: array-like
    :return: list of segment indices
    :rtype: list
    """
    # Get non-zero indices
    non_zero_indices = np.flatnonzero(sum_spec)
    if len(non_zero_indices) == 0:
        return []  # Return an empty list if there are no non-zero indices
    ...
```
And then matches will be estimated without failing: 

<img src="https://github.com/Karsa-Oy/Mascope/assets/162983707/03ce5d3e-4f86-4cbe-846a-727b2b851250" width=300/>

**Comment by oskarsari** [comment]
Fixed in commit [dde8f6c](https://github.com/Karsa-Oy/Mascope/commit/dde8f6c4a630c518d5cb20f01768bee719f516bf)
